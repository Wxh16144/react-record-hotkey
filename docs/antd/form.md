---
title: 搭配表单
order: 1
---

## 搭配 Ant Design Form

只需要稍微加一点代码即可搭配 antd 表单组件使用，下面示例提供了快捷键设置冲突校验。

<code src="../examples/antd-form.tsx"></code>
