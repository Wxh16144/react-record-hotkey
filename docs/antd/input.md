---
title: 基础使用
order: 0
nav:
  title: Antd Record
  order: 3
---

<code src="../../examples/with-antd5/src/App.tsx"></code>

### Interface

<<< ../../packages/antd-record-hotkey-input/src/type.ts
