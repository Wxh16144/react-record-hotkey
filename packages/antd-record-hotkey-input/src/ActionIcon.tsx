import type { ButtonProps, TooltipProps } from 'antd';
import { Button, Tooltip } from 'antd';
import type { CSSProperties } from 'react';

export interface ActionIconProps extends Omit<ButtonProps, 'title'> {
  title?: TooltipProps['title'];
  cursor?: CSSProperties['cursor'];
  placement?: TooltipProps['placement'];
  icon?: ButtonProps['icon'];
}

const ActionIcon = (props: ActionIconProps) => {
  const { placement, title, icon, cursor, onClick, ...restProps } = props;
  let iconNode = <Button icon={icon} style={{ cursor }} {...restProps} onClick={onClick} />;

  if (title) {
    iconNode = (
      <Tooltip arrow={false} title={title} placement={placement}>
        {iconNode}
      </Tooltip>
    );
  }

  return iconNode;
};

export default ActionIcon;
