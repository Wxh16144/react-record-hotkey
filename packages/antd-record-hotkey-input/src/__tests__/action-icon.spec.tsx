import { fireEvent, render } from '@test/utils';
import { ConfigProvider } from 'antd';
import ActionIcon from '../ActionIcon';

const Icon = () => <svg>o</svg>;

describe('ActionIcon', () => {
  it('should render correctly', () => {
    const { container } = render(<ActionIcon icon={<Icon />} />);
    expect(container).toMatchSnapshot();
  });

  it('should render with className', () => {
    const { getByRole } = render(<ActionIcon icon={<Icon />} className="test" />);
    expect(getByRole('button')).toHaveClass('test');
  });

  it('should render with onClick', () => {
    const onClick = vi.fn();
    const { getByRole } = render(<ActionIcon icon={<Icon />} onClick={onClick} />);
    fireEvent.click(getByRole('button'));
    expect(onClick).toBeCalled();
  });

  describe('disabled', () => {
    it('should render with disabled', () => {
      const onClick = vi.fn();
      const { getByRole } = render(<ActionIcon icon={<Icon />} disabled onClick={onClick} />);
      expect(getByRole('button')).toBeDisabled();
      fireEvent.click(getByRole('button'));
      expect(onClick).not.toBeCalled();
    });

    it('should render with disabled by antd ConfigProvider', () => {
      const { getByRole } = render(
        <ConfigProvider componentDisabled>
          <ActionIcon icon={<Icon />} />,
        </ConfigProvider>,
      );
      expect(getByRole('button')).toBeDisabled();
    });
  });

  it('should render with style', () => {
    const { getByRole } = render(<ActionIcon icon={<Icon />} style={{ color: 'red' }} />);
    expect(getComputedStyle(getByRole('button')).color).toBe('rgb(255, 0, 0)');
  });

  describe('size', () => {
    it('should render', () => {
      const { getByRole } = render(<ActionIcon icon={<Icon />} size="large" />);
      expect(getByRole('button')).toHaveClass('ant-btn-lg');
      expect(getComputedStyle(getByRole('button')).maxHeight).toBeFalsy();
    });

    it.each([void 0, 'small', 'middle'])('should render with size %s', (size: any) => {
      const { getByRole } = render(<ActionIcon icon={<Icon />} size={size} />);
      expect(getComputedStyle(getByRole('button')).maxHeight).toBeTruthy();
    });

    it('overwrites max-height', () => {
      const { getByRole } = render(
        <ActionIcon icon={<Icon />} size="small" style={{ maxHeight: 100 }} />,
      );
      expect(getComputedStyle(getByRole('button')).maxHeight).toBe('100px');
    });

    it('should render with size by antd ConfigProvider', () => {
      const { getByRole } = render(
        <ConfigProvider componentSize="small">
          <ActionIcon icon={<Icon />} />,
        </ConfigProvider>,
      );
      expect(getByRole('button')).toHaveClass('ant-btn-sm');
      expect(getComputedStyle(getByRole('button')).maxHeight).toBeTruthy();
    });
  });
});
