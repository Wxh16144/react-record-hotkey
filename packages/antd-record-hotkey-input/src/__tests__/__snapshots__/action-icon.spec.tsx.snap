// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ActionIcon > should render correctly 1`] = `
<div>
  <button
    class="ant-btn ant-btn-default ant-btn-icon-only ant-action-icon"
    style="max-height: 22px;"
    type="button"
  >
    <span
      class="ant-btn-icon"
    >
      <svg>
        o
      </svg>
    </span>
  </button>
</div>
`;
